### PR TITLE
Remove fractional snapshot calculations

### PR DESCRIPTION
**Pull request by peer2f00l**
Significantly simplifies code and reduces edge-cases.

### CODE REVIEW

File: contract/market/tests/snapshot.rs [hunk 1]
```diff
@@ -278,17 +268,15 @@ async fn snapshot_with_full_repayment() {
     eprintln!("Total liability before repayment: {total_liability:?}");
 
     // Repay everything (including any accrued interest)
-    c.repay(&borrow_user, total_liability).await;
+    let r = c.repay(&borrow_user, total_liability).await;
+    print_execution(&r);
```
Comment (royalf00l) [inline contract/market/tests/snapshot.rs]: are the prints still intentional?

File: contract/market/tests/snapshot.rs [hunk 2]
```diff
@@ -301,6 +289,9 @@ async fn snapshot_with_full_repayment() {
         final_position.get_total_borrow_asset_liability()
     );
 
+    eprintln!("Final snapshot:");
```
Comment (royalf00l) [inline contract/market/tests/snapshot.rs:291]: do we still want these 2 prints?

Comment (royalf00l) [reply contract/market/tests/snapshot.rs:291]: I guess these are tests, so not a big deal, just double checking

File: common/src/borrow.rs [hunk 3]
```diff
@@ -475,52 +493,117 @@ impl<'a> BorrowPositionGuard<'a> {
...
+            //  more research to see if it is possible for the signature to
+            //  still show up on chain after the promise expires.
+            //
+            // Mitigation strategy: Retain locks until we know the
+            // signature will not be issued. Note that we can't implement
+            // this strategy until we implement asset transfer for MPC
+            // assets, so we IGNORE THIS CASE FOR NOW.
+            //
+            // TODO: Implement case 2 mitigation.
+            // NOTE: Not needed for chain-local (NEP-141-only) tokens.
```
Comment (royalf00l) [inline common/src/borrow.rs:605]: Empty else case. No code implemented here

Comment (peer2f00l) [reply common/src/borrow.rs:605]: It's here for the comment, which I think is important to acknowledge.

File: contract/market/tests/snapshot.rs (hunk 1 above)
Comment (peer2f00l) [reply contract/market/tests/snapshot.rs]: this one is unnecessary to keep around

File: contract/market/tests/snapshot.rs (hunk 2 above)
Comment (peer2f00l) [reply contract/market/tests/snapshot.rs:291]: yeah I think these ones are ok to keep

**Review by royalf00l — APPROVED** [review]
